### DISCUSSION TITLE
Clustering

### DISCUSSION THREAD

**Discussion by skinkie** [halfgaar/FlashMQ#56] [Ideas]
While the architecture of FlashMQ scales well, I wonder what the path to clustering would look like. Is there an active server envisioned where publications end up, and slaves that take the (read-only) subscriptions? I think it can already be created by a client that subscribes on the active server, and publishes on slaves. Is there any design present for future clustering?

**Comment by halfgaar** [comment]
The way you suggest still has bottle necks. What would probably be required, is that servers are aware of each others clients and subscriptions. Some kind of meta-data link between the servers would be required. Currently no plans are underway though.

However, _bridge mode_ is actually being worked on. That's not quite clustering, but at least provides some ability for sending messages other servers.
